File: 2022_python/220327.py
#
# created by z on 3/27/2022
# 1337

class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:

        s: List[List[int]] = [[0 ] *2 for _ in range(len(mat))] # python is weird
        for i in range(len(mat)):
            num: int = 0
            for j in range(len(mat[0])):
                if mat[i][j] == 1:
                    num += 1
                else:
                    break
            s[i][0] = num
            s[i][1] = i

        # There's no way to compare two or more values in one go in python3
        # The solution is to sort two times

        s.sort(key=lambda x: x[1])
        s.sort(key=lambda x: x[0])

        ret: List[int] = [0] * k
        for i in range(k):
            ret[i] = s[i][1]
        return ret

File: 2022/2022_03_cpp/220327.cpp
//
// Created by z on 3/27/2022.
// # 1337

class Solution {
public:

    typedef pair<int, int> p;

    static bool cmp (p a, p b)
    {
        return a.first != b.first ? a.first < b.first : a.second < b.second;
    }


    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        vector<p> soldiers(mat.size(), p(0,0));
        for (int i = 0; i < mat.size(); i++)
        {
            int num = 0;
            for (int j = 0; j < mat[0].size(); j++)
            {
                if (mat[i][j] == 1)
                    num++;
                else
                    break;
            }
            soldiers[i].first  = num;
            soldiers[i].second = i;
        }
        sort(soldiers.begin(), soldiers.end(), cmp);

        vector<int> ret;
        for (int i = 0; i < k; i++)
        {
            ret.push_back(soldiers[i].second);
        }
        return ret;
    }
};